src/filesystem.js

```diff
@@ -330,6 +331,14 @@ export default class Filesystem extends EventEmitter {
               : result;
           });
       }
+    } else if (method === 'archive') {
+      const [selection] = args;
+      const mount = this.getMountpointFromPath(selection[0].path);
```

Comment:
This will throw an exception if somehow an empty selection is supplied :thinking: 

Probably best to handle this.

Also, this condition can be removed and the stuff below can be changed to (excluding this check):

```js
    const [input] = args;
    const file = method === 'archive' ? input[0] : input;
    const mount = this.getMountpointFromPath(file);
    if (!mount) {
      throw new Error('Failed to resolve mountpoint');
    }
    return VFS[method](mount._adapter, mount)(...args);
```